skythought/skythought_evals/inference_and_check.py

```diff
@@ -324,7 +352,24 @@ def perform_inference_and_save(
     )
 
     for temp in temperatures:
-        if args.model.startswith("openai"):
+        if len(conversations) == 0:
+            print("No more data to process")
+            continue
+        elif args.use_ray:
+            config = load_config_from_path(args.ray_config)
+            config["model_id"] = args.model
+            engine_cfg = init_engine_from_config(config)
+            ds = ray.data.from_items([(idx, conv) for idx, conv in enumerate(conversations)])
+            if config.get("num_replicas", 1) > 1 and config.get("num_replicas", 1) > ds.num_blocks():
+                ds = ds.repartition(num_partitions=config["num_replicas"])
+            ds = EvalWorkload(dataset=ds, sampling_params={"n": args.n, "max_tokens": max_tokens, "temperature": temp})
+            pipeline = Pipeline(
+                engine_cfg,
+                env_config=EnvConfig(**config["env_config"]),
+            )
+            ds = pipeline(ds)
```

Comment:
wait, this should be ds = pipeline.run(). ds is not a dataset here, it's a workload object. Please be careful with naming. 


```python
workload = EvalWorkload(dataset=ds, sampling_params={"n": args.n, "max_tokens": max_tokens, "temperature": temp})

pipeline = Pipeline(...)
ds = pipeline.run(workload)
ds = ds.materialize()
responses = [res for res in ds.iter_rows()]
```